src/main/java/com/google/step/snippet/external/GeeksForGeeksClient.java

```diff
@@ -51,9 +56,28 @@ public Card search(String geeksLink) {
     if (snippets.isEmpty() || snippets.first().getElementsByClass(CODE_CLASS).text().isEmpty()) {
       return null;
     }
-    String title = titles.first().text();
-    String description = descriptions.first().text();
+    String title = Jsoup.clean(titles.first().text(), Whitelist.relaxed());
+    String description = Jsoup.clean(descriptions.first().text(), Whitelist.relaxed());
     String code = snippets.first().getElementsByClass(CODE_CLASS).text();
+    if (containsEscape(query.toLowerCase())
+        || containsEscape(geeksLink)
+        || containsEscape(title.toLowerCase())
+        || containsEscape(description.toLowerCase())) {
+      title = StringEscapeUtils.escapeHtml4(title);
+      description = StringEscapeUtils.escapeHtml4(description);
+      code = StringEscapeUtils.escapeHtml4(code);
+    } else {
+      code = Jsoup.clean(code, Whitelist.relaxed().preserveRelativeLinks(true));
+    }
     return new Card(title, code, geeksLink, description, SOURCE_NAME, ICON_LINK);
   }
+
+  private boolean containsEscape(String possibleHtml) {
```

Comment:
I'm going to expect a future PR to move this to the Client abstract class (in this PR, it's still an interface).